test/board_test.rb

```diff
@@ -0,0 +1,201 @@
+    assert_equal false, @board.coordinates_consecutive?(["A1", "C1"])
+    assert_equal false, @board.coordinates_consecutive?(["A3", "A2", "A1"])
+    assert_equal false, @board.coordinates_consecutive?(["C1", "B1"])
+  end
+
+  def test_coordinates_must_be_same_length_as_ship
+    assert_equal false, @board.valid_placement?(@cruiser, ["A1", "A2"])
+    assert_equal false, @board.valid_placement?(@submarine, ["A2", "A3", "A4"])
+
+    assert_equal true, @board.valid_placement?(@cruiser, ["A1", "A2", "A3"])
+    assert_equal true, @board.valid_placement?(@submarine, ["A2", "A3"])
+  end
+
+  def test_coordinates_must_be_consecutive
+    assert_equal false, @board.valid_placement?(@cruiser, ["A1", "A2", "A4"])
+    assert_equal false, @board.valid_placement?(@submarine, ["A1", "C1"])
+    assert_equal false, @board.valid_placement?(@cruiser, ["A3", "A2", "A1"])
+    assert_equal false, @board.valid_placement?(@submarine, ["C1", "B1"])
+
+    assert_equal true, @board.valid_placement?(@cruiser, ["A1", "A2", "A3"])
+    assert_equal true, @board.valid_placement?(@cruiser, ["A1", "B1", "C1"])
+    assert_equal true, @board.valid_placement?(@submarine, ["C1", "D1"])
+    assert_equal true, @board.valid_placement?(@submarine, ["D1", "D2"])
+  end
+
+  def test_coordinates_cant_be_diagonal
+
+    assert_equal false, @board.valid_placement?(@cruiser, ["A1", "B2", "C3"])
+    assert_equal false, @board.valid_placement?(@submarine, ["C2", "D3"])
+  end
+
+  def test_it_can_use_all_criteria_to_validate_placement
+    assert_equal true, @board.valid_placement?(@submarine, ["A1", "A2"])
+    assert_equal true, @board.valid_placement?(@cruiser, ["B1", "C1", "D1"])
+  end
+
+  def test_ships_occupy_multiple_cells
+    @board.place(@cruiser, ["A1", "A2", "A3"])
+    cell_1 = @board.cells["A1"]
+    cell_2 = @board.cells["A2"]
+    cell_3 = @board.cells["A3"]
+
+    assert_equal @cruiser, cell_1.ship
+    assert_equal @cruiser, cell_2.ship
+    assert_equal @cruiser, cell_3.ship
+  end
+
+  def test_if_all_cells_are_empty
+    @board.place(@cruiser, ["A1", "A2", "A3"])
+    assert_equal true, @board.cell_has_no_ships?(["B2", "B3", "B4"])
+    assert_equal false, @board.cell_has_no_ships?(["A1","A2","A3"])
+  end
+
+  def test_ships_do_not_overlap
+    cell_1 = @board.cells["A1"]
+    cell_2 = @board.cells["A2"]
+    cell_3 = @board.cells["A3"]
+    @board.place(@cruiser, ["A1", "A2", "A3"])
+    @board.place(@submarine, ["A1", "A2"])
+
+    assert_equal @cruiser, cell_1.ship
+    assert_equal @cruiser, cell_2.ship
+    assert_equal @cruiser, cell_3.ship
+    assert_equal false, @board.valid_placement?(@submarine, ["A1", "B1"])
+  end
+
+  def test_it_can_render_default
+    cell_1 = @board.cells["A1"]
+    cell_2 = @board.cells["A2"]
+    cell_3 = @board.cells["A3"]
+
+    assert_equal ".", cell_1.render
+    assert_equal ".", cell_2.render
+    assert_equal ".", cell_3.render
+  end
+
+  def test_it_can_render_ship_visible
```

Comment:
I suggest we implement the above assertion structure for the rest of these tests as well, instead of testing the individual cells so that we can better test the full board.

Or, [this](https://github.com/arpariseau/battleship/blob/master/test/board_test.rb) repo seems to use another good alternative for testing the rendered boards, if we want to try something like that instead.

Not sure which one would be a better approach, but my instinct is to use the one that allows us to confirm as much as possible with making as few assumptions as possible about what we're not able to test.